core/gui/src/app/workspace/component/left-panel/versions-list/versions-list.component.html

```diff
@@ -30,7 +32,7 @@
           <button
             nz-button
             nzType="link"
-            (click)="getVersion(row.vId)"
+            (click)="getVersion(row.vId, (l - i), i)"
```

Comment:
Please add a public method that returns a value (l - i) and use that method instead of directly computing it here.
Replace all of the (l - i) to that public method.